lms/views/application_instances.py

```diff
@@ -10,15 +10,14 @@
 )
 def create_application_instance(request):
     """Create application instance in the databse and respond with key and secret."""
-    # Check to make sure that both developer key and secret are present. If not set to None.
+
+    # Default developer_key and developer_secret to None rather than letting
+    # them be empty strings.
     developer_key = request.params["developer_key"].strip()
     developer_secret = request.params["developer_secret"].strip()
-    if developer_key == "":
-        developer_key = None
-
-    if developer_secret == "":
-        developer_secret = None
```

Comment:
If either of these is falsey (e.g. an empty string) then they both get set to `None` right below, so these four lines are unnecessary. (The fact that it silently deletes the given value for either one of these fields if the other field is empty doesn't seem great but I'm not fixing this page right now, just adding test coverage.)